### PR TITLE
Continue updating packaging/PC docs

### PR DESCRIPTION
**Pull request by gguillotte**
This supports work to eliminate the unversioned platforms and system requirements guides.

-   Update style and formatting on agent life cycle include and platforms guide.
-   Convert headings to sentence case.
-   Standardize common text in per-platform install docs.
-   Add information about Puppet Collections to per-platform install docs.
-   Add Puppet 4.4 PC1 includes for single-sourcing.
-   Use ordered lists when instructions consist of a list of steps.
-   As a temporary improvement, update formatting and brand references on the unversioned platforms guide.

### CODE REVIEW

File: source/_includes/pup44_platforms_windows.markdown [hunk 1]
```diff
@@ -0,0 +1,16 @@
+We publish and test official [`puppet-agent`](/puppet/latest/reference/about_agent.html) packages for the following versions of Windows:
+
+-   Windows Server 2012 R2 (supported by [Puppet Enterprise][peinstall])
+-   Windows Server 2008 R2 (supported by [Puppet Enterprise][peinstall])
+
+We also publish, but do not automatically test, `puppet-agent` packages for the following versions of Windows:
+
+-   Windows Server 2012 (supported by [Puppet Enterprise][peinstall])
+-   Windows Server 2008 (supported by [Puppet Enterprise][peinstall])
+-   Windows 8.1 (supported by [Puppet Enterprise][peinstall])
```
Comment (gguillotte) [inline source/_includes/pup44_platforms_windows.markdown]: Add Windows 10.

File: source/guides/platforms.markdown [hunk 2]
```diff
@@ -39,27 +34,23 @@ Linux
...
 
-Other Unix
-----------
+## Other Unix
 
-- Mac OS X, version 10.7 (Leopard) and higher
+- OS X, version 10.7 (Leopard) and higher
 - Oracle Solaris, version 10 and higher
 - AIX, version 5.3 and higher
 - HP-UX
```
Comment (gguillotte) [inline source/guides/platforms.markdown]: Confirm all above this line.

File: source/puppet/4.4/reference/system_requirements.markdown [hunk 3]
```diff
@@ -4,87 +4,78 @@ title: "Puppet 4.4 System Requirements"
...
 
-* Oracle Solaris, version 10 and higher (Puppet performs limited automated testing on Solaris 11)
-* AIX, version 5.3 and higher
-* FreeBSD 4.7 and later
-* OpenBSD 4.1 and later
-* HP-UX
+> **Note:** For platforms supported in Puppet Enterprise, see its [System Requirements][pe-requirements].
 
-## Basic Requirements
+Puppet also makes the `puppet-agent` packaging process available as an open source project. For information about building the package yourself, see the [`puppetlabs/puppet-agent` repository on GitHub](https://github.com/puppetlabs/puppet-agent).
```
Comment (gguillotte) [inline source/puppet/4.4/reference/system_requirements.markdown]: Confirm that we can recommend the `puppet-agent` repo to open source users who want to build from source, as there isn't a guide to build Puppet 4 from source.

File: source/puppet/4.4/reference/system_requirements.markdown [hunk 4]
```diff
@@ -4,87 +4,78 @@ title: "Puppet 4.4 System Requirements"
...
 
-If you're installing Puppet via the official packages, you won't need to worry about these prerequisites; your system's package manager handles all of them. These are only listed for those running Puppet from source or on unsupported systems.
+### Prerequisites
 
-Puppet 4.4 has the following prerequisites:
+The official `puppet-agent` package either includes or instructs your operating system's package manager to install all prerequisites, including its own Ruby interpreter. When running Puppet from source, building your own `puppet-agent` package, or otherwise running Puppet binaries without installing the official package, Puppet services have the following prerequisites:
```
Comment (gguillotte) [inline source/puppet/4.4/reference/system_requirements.markdown]: Confirm this section.

File: source/puppet/4.4/reference/system_requirements.markdown [hunk 5]
```diff
@@ -4,87 +4,78 @@ title: "Puppet 4.4 System Requirements"
...
 
-The Puppet agent service has no particular hardware requirements and can run on nearly anything.
+> **Note:** For [Puppet Enterprise's](/pe/) supported platforms, see its [system requirements][pe-requirements].
 
-However, the Puppet master service is fairly resource intensive, and should be installed on a robust dedicated server.
+## Puppet agent
 
-* At a minimum, your Puppet master server should have two processor cores and at least 1 GB of RAM.
-* To comfortably serve at least 1,000 nodes, it should have 2-4 processor cores and at least 4 GB of RAM.
+The Puppet agent service is lightweight enough to run on nearly any system capable of running a Ruby interpreter. To be managed by a Puppet master, the agent must be accessible from the master. See the [pre-installation instructions][preinstall-network] for details about the required DNS and port configuration for agents.
```
Comment (gguillotte) [inline source/puppet/4.4/reference/system_requirements.markdown]: Confirm. Specifying precise requirements for Puppet agent would resolve several tickets and requests.

File: source/puppet/4.4/reference/system_requirements.markdown [hunk 6]
```diff
@@ -4,87 +4,78 @@ title: "Puppet 4.4 System Requirements"
...
 
-The demands on the Puppet master vary widely between deployments. The total needs are affected by the number of agents being served, how frequently those agents check in, how many resources are being managed on each agent, and the complexity of the manifests and modules in use.
-
-{% include agent_lifecycle.md %}
-
-## Platforms With Packages
-
-Puppet 4.4 and all of its prerequisites run on the following platforms, and Puppet provides official packages in [Puppet Collections](./upgrade_minor.html#puppet-collections-and-upgrading).
+For simple installation, Puppet builds and packages the open source agent and its prerequisites for 32-bit (i386) and 64-bit (x86_64 and amd64) versions of many Linux distributions, OS X, and Windows.
```
Comment (gguillotte) [inline source/puppet/4.4/reference/system_requirements.markdown]: Confirm. OS X packages appear to be 64-bit only.

File: source/puppet/4.4/reference/install_osx.md [hunk 7]
```diff
@@ -29,94 +38,119 @@ To install on other operating systems, see the pages linked in the navigation si
...
-These packages are tied to Puppet Collection 1, which is a set of Puppet software designed to work well with Puppet 4. The `puppet-agent` package bundles all of Puppet's prerequisites, so you don't need to download anything else to install Puppet on an agent node.
+You can also download older versions of Puppet from [downloads.puppetlabs.com](https://downloads.puppetlabs.com/mac/). Browse to `<OS X VERSION>/PC1/x86_64` for the most recently released packages. For example, the path to the PC1 package for OS X 10.11 is `https://downloads.puppetlabs.com/mac/10.11/PC1/x86_64`.
 
 ### Choosing a package
 
-OS X packages are named according to their `puppet-agent` version and compatible OS X version:
+Packages are named according to their `puppet-agent` version and compatible OS X version. They also contain a release number --- typically `1` --- that represents the iteration of that version's package.
 
-    puppet-agent-<PACKAGE VERSION>.osx<OS X VERSION>.dmg
+    puppet-agent-<PACKAGE VERSION>-<RELEASE NUMBER>.osx<OS X VERSION>.dmg
```
Comment (gguillotte) [inline source/puppet/4.4/reference/install_osx.md]: Per @melissa, all OS X packages now append a release number to the version number. This note needs to be backported to all Puppet docs relevant to `puppet-agent` since 1.2.4.

File: source/puppet/4.4/reference/install_osx.md [hunk 8]
```diff
@@ -29,94 +38,119 @@ To install on other operating systems, see the pages linked in the navigation si
 {% include puppet-collections/_puppet_collection_1_osx1010.md %}
 {% include puppet-collections/_puppet_collection_1_osx1009.md %}
 
-You can also download [older versions of Puppet](https://downloads.puppetlabs.com/mac/); browse to `<OS X VERSION>/PC1/x86_64` for the most recently released packages.
-
-These packages are tied to Puppet Collection 1, which is a set of Puppet software designed to work well with Puppet 4. The `puppet-agent` package bundles all of Puppet's prerequisites, so you don't need to download anything else to install Puppet on an agent node.
+You can also download older versions of Puppet from [downloads.puppetlabs.com](https://downloads.puppetlabs.com/mac/). Browse to `<OS X VERSION>/PC1/x86_64` for the most recently released packages. For example, the path to the PC1 package for OS X 10.11 is `https://downloads.puppetlabs.com/mac/10.11/PC1/x86_64`.
```
Comment (gguillotte) [inline source/puppet/4.4/reference/install_osx.md]: The `downloads.puppetlabs.com` URL is intentional. It was not changed during the rebrand.

File: source/puppet/4.4/reference/install_osx.md [hunk 9]
```diff
@@ -29,94 +38,119 @@ To install on other operating systems, see the pages linked in the navigation si
...
-    sudo hdiutil unmount /Volumes/<IMAGE>
+        sudo hdiutil unmount /Volumes/<IMAGE>
 
-You can then delete the `.dmg` file.
+    You can then delete the disk image.
 
 ### Upgrading with Puppet
 
-Puppet includes a `package` resource provider for OS X that can install `.pkg` files from a disk image. If you already have Puppet installed, you can use the `puppet resource` command to upgrade with fewer steps.
+Puppet includes a [`pkgdmg` package provider](./type.html#package-provider-pkgdmg) that can install OS X installation packages (`.pkg` files) from a disk image (`.dmg` file). If you already have Puppet installed, you can use the `puppet resource` command to upgrade with fewer steps.
```
Comment (gguillotte) [inline source/puppet/4.4/reference/install_osx.md]: Confirm.

**Comment by gguillotte** [comment]
CC @jtappa @nfagerlund @kylog 

@jtappa If this conflicts with things you're working on, please let me know and I can refactor this PR or close it and wait for your work to be merged.

File: source/puppet/4.4/reference/system_requirements.markdown (hunk 3 above)
Comment (melissa) [inline source/puppet/4.4/reference/system_requirements.markdown]: 👍  this should be fine to recommend to users. We do need more documentation on building from source and there's also work to be done to make it easier for the community. But that is all still in progress

File: source/puppet/4.4/reference/system_requirements.markdown (hunk 4 above)
Comment (melissa) [inline source/puppet/4.4/reference/system_requirements.markdown]: The packages do not `instruct` any installation. There should be no dependencies installed alongside the puppet-agent package at install time. Most of the required bits outlined below are already included in the packages we provide. `rgen` is vendored as a part of puppet, `ruby`, `hiera`, and `facter` are packaged up in puppet-agent. So the user doesn't have much of a choice in terms of the version of these libraries and definitely doesn't need them on the system prior to installation of the puppet-agent package.

`json` and `msgpack` I actually can't find, so I'd like to get someone else to confirm how/if those are provided.

File: source/puppet/4.4/reference/install_osx.md (hunk 7 above)
Comment (melissa) [inline source/puppet/4.4/reference/install_osx.md]: 👍

**Comment by jtappa** [comment]
@gguillotte no conflicts for me

File: source/puppet/4.4/reference/system_requirements.markdown (hunk 4 above)
Comment (gguillotte) [inline source/puppet/4.4/reference/system_requirements.markdown]: @melissa @kylog Hey, can I get a confirmation on this before we merge this head of the next release? Or should I revert this change?

File: source/puppet/4.4/reference/install_osx.md (hunk 9 above)
Comment (gguillotte) [inline source/puppet/4.4/reference/install_osx.md]: @kylog The `package` resource docs suggest `pkgdmg` is the appropriate provider here, so is this change from `package` appropriate?

File: source/puppet/4.4/reference/system_requirements.markdown (hunk 5 above)
Comment (gguillotte) [inline source/puppet/4.4/reference/system_requirements.markdown]: @kylog Do we have official system requirements for Puppet agent?

Also, should we continue to note requirements for the Ruby master, and if so, can I confirm what they are?

**Comment by gguillotte** [comment]
@jtappa This is pretty much ready to go, save for a couple of questions for the Puppet team on whether to refer to `pkgdmg`, the system requirements for agent and Ruby master, and the prereqs for building from source.

File: source/puppet/4.4/reference/system_requirements.markdown (hunk 5 above)
Comment (kylog) [inline source/puppet/4.4/reference/system_requirements.markdown]: We don't have official system requirements for Puppet agent. I agree with removing system requirements for puppet master *here*. I'm assuming they exist somewhere for puppetserver, though?

Also "the agent must be accessible from the master" implies that the master reaches out to the agent which is not the case. I might rephrase as "the agent must be able to access the master at port 8140".

File: source/puppet/4.4/reference/system_requirements.markdown (hunk 4 above)
Comment (kylog) [inline source/puppet/4.4/reference/system_requirements.markdown]: Seems okay to me. `json` is included with ruby itself now and `msgpack` isn't a requirement (it's optional).

File: source/puppet/4.4/reference/install_osx.md (hunk 9 above)
Comment (kylog) [inline source/puppet/4.4/reference/install_osx.md]: I'd be tempted to drop that whole sentence. At this point, I don't think the reader needs to know what providers do what. (And it doesn't seem like the other `puppet resource` examples in this set of pages go into which provider will be doing the work for them.

**Comment by gguillotte** [comment]
Once this is merged, I'll follow this PR up with the changes for 4.5.
